lib/datadog/profiling/component.rb

```diff
@@ -111,19 +111,30 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
       end
 
       private_class_method def self.enable_gc_profiling?(settings)
-        # See comments on the setting definition for more context on why it exists.
-        if settings.profiling.advanced.force_enable_gc_profiling
-          if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3')
-            Datadog.logger.debug(
-              'Profiling time/resources spent in Garbage Collection force enabled. Do not use Ractors in combination ' \
-              'with this option as profiles will be incomplete.'
-            )
-          end
+        return false unless settings.profiling.advanced.gc_enabled
```

Comment:
For allocation we moved it out of advanced. Should we apply the same reasoning to gc?

Reply:
My thinking behind putting that one out of advanced and leaving this one in advanced, is that I'm not sure exactly when (if?) we're going to enable allocation profiling by default.

Thus, it's a setting that I expect many users will want to toggle until we change that.

---

Conversely, for GC profiling, given we are _enabling_ it by default I don't think anyone will need or should to touch this.

I'd say its existence is mostly a support mechanism, where if we find any issue that we suspect may be related to the feature, we have the means to disable it.

(This is why I left `timeline_enabled` under advanced as well).

LMK your thoughts, if you still think this distinction is too subtle, I don't particularly mind moving it.

Reply:
Gotcha. I think it makes sense and just wanted to understand the pattern you were going with here so I could follow it in the future :)